include/class.role.php

```diff
@@ -320,6 +320,10 @@ function merge($perms) {
     }
 
     static function allPermissions() {
+        // Sort permissions in alphabetical order
+        foreach (static::$_permissions as $k => $v) {
+            asort(static::$_permissions[$k]);
+        }
```

Comment:
This is probably fine, but it could result in unnecessary sorting, since its sorted on every invocation. I would suggest
```php
static function allPermissions() {
    static $sorted = false;

    if (!$sorted) {
        // Sort permissions in alphabetical order
        foreach (static::$_permissions as $k => $v) {
            asort(static::$_permissions[$k]);
        }
        $sorted = true;
    }

    return static::$_permissions;
}
```